### PR TITLE
[App] Fix buy button

### PR DESCRIPTION
**Pull request by Maks19**
**Description:** fix buy button

**Issue:** https://www.notion.so/saladtech/BUY-button-inactive-830a8c06c78e4b38923a79f7ae748585

### CODE REVIEW

File: packages/web-app/src/modules/reward/RewardStore.tsx [hunk 1]
```diff
@@ -211,20 +208,13 @@ export class RewardStore {
...
-    }
-
     //Ensures that the user is logged in
     try {
       yield this.store.auth.login()
     } catch {
       return
     }
 
-    this.isRedeeming = true
```
Comment (scottcinatl) [inline packages/web-app/src/modules/reward/RewardStore.tsx:226]: I would move this update of the `isRedeeming` state to the start of the `try` block below.

**Review by scottcinatl — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/salad-pay/SaladPay.tsx [hunk 2]
```diff
@@ -12,13 +12,6 @@ export class SaladPay {
   paymentRequest = (options: SaladPaymentRequestOptions): SaladPaymentRequest => {
     console.log('Creating a new payment request')
 
-    // Check if there is already one and abort previous ones
```
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/salad-pay/SaladPay.tsx]: I left this functionality but split abort and hide logic

File: packages/web-app/src/modules/reward/RewardStore.tsx [hunk 3]
```diff
@@ -211,20 +208,13 @@ export class RewardStore {
       this.lastRewardId = reward.id
     }
 
-    if (this.isRedeeming) {
```
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/reward/RewardStore.tsx:214]: I think `this.processing` is preventing multiple Buy button clicks from occurring. So I'm not sure for what this prop can be using now
https://github.com/SaladTechnologies/salad-applications/blob/master/packages/web-app/src/modules/salad-pay/SaladPayStore.tsx#L42

File: packages/web-app/src/modules/salad-pay/SaladPay.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/salad-pay/SaladPay.tsx]: Fixed

File: packages/web-app/src/modules/reward/RewardStore.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/reward/RewardStore.tsx]: Fixed

**Pull request merged**